### PR TITLE
Rename DomainCreator to the more accurate ZoneCreator

### PR DESCRIPTION
**Pull request by Yannik**
`DomainCreator` and `EnsureDomainExists` are misleading, because it actually creates zones, not domains.

### CODE REVIEW

**Comment by tlimoncelli** [comment]
Thanks for fixing this inconsistency.  Please also fix the comments and other strings that surround these changes.

**Comment by Yannik** [comment]
@tlimoncelli done.

**Review by tlimoncelli — APPROVED** [review]

File: commands/previewPush.go [hunk 1]
```diff
@@ -154,9 +154,9 @@ DomainLoop:
 						out.Warnf("Domain '%s' does not exist in the '%s' profile and will be added automatically.\n", domain.Name, provider.Name)
```
Comment (tlimoncelli) [inline commands/previewPush.go:154]: Change "Domain" to "Zone" in out.Warnf

File: providers/dnsmadeeasy/dnsMadeEasyProvider.go [hunk 2]
```diff
@@ -177,8 +177,8 @@ func (api *dnsMadeEasyProvider) GetDomainCorrections(dc *models.DomainConfig) ([
 	return corrections, nil
 }
 
-// EnsureDomainExists returns an error if domain doesn't exist.
-func (api *dnsMadeEasyProvider) EnsureDomainExists(domainName string) error {
+// EnsureZoneExists returns an error if domain doesn't exist.
+func (api *dnsMadeEasyProvider) EnsureZoneExists(domainName string) error {
```
Comment (tlimoncelli) [inline providers/dnsmadeeasy/dnsMadeEasyProvider.go:181]: domainName should be zoneName.

If we're going to make this change, we might as well change the "domain string" parameter in most of these other functions. I wouldn't change much more (like the api.domainExists() call below)

Comment (Yannik) [reply providers/dnsmadeeasy/dnsMadeEasyProvider.go:181]: done

File: commands/previewPush.go (hunk 1 above)
Comment (Yannik) [reply commands/previewPush.go]: done

**Comment by Yannik** [comment]
@tlimoncelli I have resolved the points you have raised.

**Comment by Yannik** [comment]
I have reverted the `domain->zoneName` renaming. IMO, "Creating a zone for a domain" is quite accurate. Additionally, this creates major inconsistencies with other variable names.

**Pull request merged**